site/content/guides/monitor-many-urls.md

```diff
@@ -0,0 +1,204 @@
+
+function downloadSitemap(url: string): Promise<string> {
+  return new Promise((resolve, reject) => {
+    https.get(url, (response) => {
+      let data = '';
+      
+      response.on('data', (chunk) => {
+        data += chunk;
+      });
+      
+      response.on('end', () => {
+        resolve(data);
+      });
+      
+      response.on('error', (error) => {
+        reject(error);
+      });
+    }).on('error', (error) => {
+      reject(error);
+    });
+  });
+}
+
+function parseSitemap(xmlContent: string): string[] {
+  const urls: string[] = [];
+  // Simple regex to extract URLs from <loc> tags
+  // this should work for most sitemaps but isn't universal.
+  const locRegex = /<loc>(.*?)<\/loc>/g;
+  let match;
+  
+  while ((match = locRegex.exec(xmlContent)) !== null) {
+    urls.push(match[1]);
+  }
+  
+  return urls;
+}
+
+console.log(`Downloading sitemap from: ${sitemapUrl}`);
+
+try {
+  const xmlContent = await downloadSitemap(sitemapUrl);
+  console.log('Sitemap downloaded successfully');
+  
+  const sitemapUrls = parseSitemap(xmlContent);
+  console.log(`Found ${sitemapUrls.length} URLs in sitemap`);
+  console.log('\nSitemap URLs:');
+  console.log(sitemapUrls);
+
+  //create paths and friendly names for each monitor
+  sitemapUrls.forEach((url, index) => {
+    const urlPath = new URL(url).pathname.replace(/\//g, '-').replace(/^-+|-+$/g, '') || 'root'
+    const monitorId = `checkly-${urlPath}-ping`
+    const monitorName = `Checkly ${urlPath.replace(/-/g, ' ')} Monitor`
+
+    //create each monitor with a five minute interval
+    new UrlMonitor(monitorId, {
+      frequency: Frequency.EVERY_5M,
+      name: monitorName,
+      activated: true,
+      request: {
+        url: url,
+        skipSSL: false,
+        followRedirects: true,
+        assertions: [
+          UrlAssertionBuilder.statusCode().equals(200),
+        ]
+      }
+    })
+  })
+} catch (error) {
+  console.error('Error:', error);
+  process.exit(1);
+}
+```
+
+Let's test these URL monitors and preview their deployment with:
+
+```bash
+npx checkly test && npx checkly deploy -p
+```
+
+If you're using the sitemap URL above (for a list of OpenAI webinar pages), you'll get a short list of monitors that have been tested:
+
+![A terminal](/guides/images/monitor-many-urls-02.png)
+
+Followed by the deployment preview:
+
+![A terminal](/guides/images/monitor-many-urls-03.png)
+
+If these look correct, you’re ready to run `npx checkly deploy` and create your new monitors!
+
+## Next Steps: Grouping, and More Advanced Checks
+
+With all of our monitors managed with a single file, we can update their configuration just by changing the values in the `new UrlMonitor` block above. However, as we have more detailed and nuanced monitoring, it would be nice to have a single place to manage configuration for timing, alerting, and notification channels for a group of checks. All of that is possible with Checkly Groups.
```

Comment:
I agree with Laura, we should either link to the groups construct here or add an example code snippet below. Otherwise its a little unclear how to act on this.